torch/csrc/autograd/function.cpp

```diff
@@ -47,7 +47,8 @@ variable_list Function::traced_apply(variable_list inputs) {
 
   // Finally apply this Function.
   state_lock.unlock();
-  variable_list outputs = apply(inputs);
+  variable_list copied_inputs = inputs;
+  variable_list outputs = apply(std::move(copied_inputs));
```

Comment:
Why do we need to copy them?

Reply:
inputs can be used later in the function.